### PR TITLE
objstore: Added support for passing YAML content through the flag.

### PR DESCRIPTION
**Pull request by bwplotka**
Fixes https://github.com/improbable-eng/thanos/issues/539

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: cmd/thanos/flags.go [hunk 1]
```diff
@@ -124,3 +125,46 @@ func modelDuration(flags *kingpin.FlagClause) *model.Duration {
 
 	return value
 }
+
+type pathOrContent struct {
```
Comment (domgreen) [inline cmd/thanos/flags.go]: would be tempted to call this `storeConfig`

File: cmd/thanos/flags.go [hunk 2]
```diff
@@ -124,3 +125,46 @@ func modelDuration(flags *kingpin.FlagClause) *model.Duration {
...
+		PlaceHolder("<bucket.config-yaml-path>").String()
+
+	bucketConf := cmd.Flag(fmt.Sprintf("objstore%s.config", suffix), fmt.Sprintf("Alternative to '%s' flag. Object store%s configuration in YAML.", fileFlagName, suffix)).
+		PlaceHolder("<bucket.config-yaml>").String()
+
+	return &pathOrContent{
+		name: fmt.Sprintf("objstore%s.config", suffix),
+
+		path:    bucketConfFile,
+		content: bucketConf,
```
Comment (domgreen) [inline cmd/thanos/flags.go]: we could just populate `content` here even if a path is passed ... means we can get rid of `Content()` and just have it as a public property.

File: cmd/thanos/flags.go (hunk 1 above)
Comment (bwplotka) [reply cmd/thanos/flags.go]: Not really - it's not tied to store in any way.

File: cmd/thanos/flags.go (hunk 2 above)
Comment (bwplotka) [reply cmd/thanos/flags.go]: We cannot. (: This is because this flag is not filled in the time of defining it

**Pull request merged**